Lib/test/test_mimetypes.py

```diff
@@ -305,6 +306,24 @@ def test_keywords_args_api(self):
         self.assertEqual(self.db.guess_extension(
             type='image/jpg', strict=False), '.jpg')
 
+    def test_added_types_are_used(self):
+        mime_type, _ = mimetypes.guess_type('test.myext')
+        self.assertEqual(None, mime_type)
+        mimetypes.add_type('testing/type', '.myext')
+        mime_type, _ = mimetypes.guess_type('test.myext')
+        self.assertEqual('testing/type', mime_type)
+
+    def test_add_type_with_undotted_extension_raises_exception(self):
```

Comment:
This test can be updated to check the warning message for an undotted extension.